### ISSUE TITLE
Remove `unsafe-eval` from `script-src`

### ISSUE THREAD

**Issue by robhudson** [mozilla/bedrock#14828]

**Comment by stevejalim** [comment]
Forgive me if this is already thought of, but the Wagtail admin may well still need `unsafe-eval` - would you mind checking and adding to the CMS-only CSP if need be?

**Comment by robhudson** [comment]
Thanks for reminding me. And yeah, I can fire up the admin and check it.

**Comment by janbrasna** [comment]
The Wagtail-specific excludes were added in #14869 — and I have just a naïve followup question — normally the excludes would be for anything under `/cms-admin/*`, save for its login screen — to safeguard the actual login surface from potential injection etc. … Is that a valid concern here, and viable to address, or it's okay to allow the unsafes & ancestors even on the login page, as currently proposed in  #14831?

**Comment by robhudson** [comment]
A PR was opened in #14831 that experimented with changing the webpack config to avoid `eval()`s for local development so we can avoid `unsafe-eval` in the CSP. This requires some extra testing and the PR went stale. Adding a note here for future explorations on this front.

**Comment by stevejalim** [comment]
I've just tried removing `script-src: unsafe-inline` from our CSP on a demo server and unfortunately this breaks some behaviour in Wagtail CMS, such as the date pickers. There's still an open issue to tackle this https://github.com/wagtail/wagtail/issues/12940

**Comment by janbrasna** [comment]
I think the idea is to exclude that hardening from `/cms-admin/*` anyways via:

https://github.com/mozilla/bedrock/blob/2ed16f38fc3e15c53ca57180ca07fef428fdfb2a/bedrock/settings/__init__.py#L215-L218

and keep the unsafes for the CMS code that needs it.

The blocker however was around Webpack settings, here: https://github.com/mozilla/bedrock/pull/14831#pullrequestreview-2512136471
_(changing source mapping settings could also affect prod — the aforementioned discussion about removing source maps even from local dev then never happened.)_

**Comment by stevejalim** [comment]
@janbrasna My take there was that dropping `unsafe-eval` was the reason the sourcemaps changed. If we only drop `unsafe-inline` for now, that might be OK, I'm thinking

**Comment by janbrasna** [comment]
Correct! (They were lumped together in the same ticket, but, alas you're absolutely right these two have different constraints and can/should be addressed separately, with different areas of concern.)

EDIT: And yes, Wagtail doesn't need unsafe-eval since v3.x, so it's only unsafe-inline for it now. (Webpack local dev source mapping is therefore the only unsafe-eval concern currently.)
